uswds/uswds-site
# USWDS-Site - Dependencies: Use overrides to remove reliance on snyk ignore

# Summary

Added an override to long standing `glob-parent` dependency. We no longer have to re-add to snyk ignore every month.

Related [documentation [gDrive 🔒]](https://docs.google.com/document/d/1RX6uYky-P37jysuBy6LEBhuMCQlMAvHU0mRJUow345o/edit#) for this process.

## Related issue

Closes #2068.

## Preview link

Preview link:
[Cloud Pages preview](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/jm-glob-parent/)

_There haven't been an visual updates. Still worth a peak and make sure everything looks good._

## Problem statement

1. Snyk keeps reporting gulp vulnerability.
2. We have to manually override and ignore every month.
3. This patches the dependency so we use the more secure version.

## Solution

1. Run clean `npm install`
2. Identify vulnerabilities with `no fix available`
3. View versions available in NPMJS
4. Add `overrides` field to `package.json`
5. Run `npm install`
6. Verify there are no vulnerabilities

I've also updated CircleCI to persist `node_modules` for more reliable builds [[ade35a7](https://github.com/uswds/uswds-site/pull/2069/commits/ade35a741c1c74eba2034a119306652afcde1038)].

The following dependencies were added to snyk ignore because they're appearing from USWDS/Compile. 
Related issue [uswds/uswds-compile#48](https://github.com/uswds/uswds-compile/issues/48#issuecomment-1523846943) 

<details><summary>@USWDS/Compile vulnerabilities</summary>

```bash
Issues with no direct upgrade or patch:
  ✗ Regular Expression Denial of Service (ReDoS) [High Severity][https://security.snyk.io/vuln/SNYK-JS-ANSIREGEX-1583908] in ansi-regex@2.1.1
    introduced by @uswds/compile@1.0.0 > gulp@4.0.2 > gulp-cli@2.3.0 > yargs@7.1.2 > string-width@1.0.2 > strip-ansi@3.0.1 > ansi-regex@2.1.1 and 4 other path(s)
  This issue was fixed in versions: 3.0.1, 4.1.1, 5.0.1, 6.0.1
  ✗ Prototype Pollution [High Severity][https://security.snyk.io/vuln/SNYK-JS-UNSETVALUE-2400660] in unset-value@1.0.0
    introduced by @uswds/compile@1.0.0 > gulp@4.0.2 > glob-watcher@5.0.5 > chokidar@2.1.8 > braces@2.3.2 > snapdragon@0.8.2 > base@0.11.2 > cache-base@1.0.1 > unset-value@1.0.0 and 30 other path(s)
  This issue was fixed in versions: 2.0.1
```
</details> 

## Major changes

Added `overrides` field in `package.json`.

## Testing and review

1. Verify `overrides` field in `package.json`
2. Run `npm install`
3. Verify there are 0 vulnerabilities
4. Run gulp tasks
5. There should be no regressions
6. CircleCI build step should run successfully

## Dependency updates

| Dependency name              | Previous version | New version |
| ---------------------------- | :--------------: | :---------: |
| `glob-parent` _via overrides_ |    -     |   6.0.2   |
